scripts/migrate-notifications-permission-check.sh

```
@@ -0,0 +1,164 @@
+  if [ ! -d "$path" ]; then
+    echo "Instance directory not found: $path"
+    exit 1
+  fi
+  migrate_instance "$path"
+else
+  # all instances
+  if [ -z "${PREFIX:-}" ]; then
+    echo "ERROR: PREFIX is not set. Aborting to avoid operating on all directories."
+    exit 1
+  fi
+  cd "$baseDirectory"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)